go/vt/vtorc/logic/tablet_discovery.go

```diff
@@ -229,6 +226,9 @@ func refreshTablets(tablets map[string]*topo.TabletInfo, query string, args []an
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
+			if tablet.Type == topodatapb.TabletType_PRIMARY && dontRefreshPrimary {
+				return
+			}
```

Comment:
I don't like this way of ignoring all the tablets that have a type PRIMARY. We should be ignoring tablet explicitly by their alias and not by the type. You can be in a situation where 2 tablets have a type PRIMARY, but we only want to ignore the one that the DeadPrimary recovery is being run on